sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/trace/OnEndSpanProcessor.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.trace;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+/** A SpanProcessor implementation that is only capable of processing spans when they end. */
+public final class OnEndSpanProcessor implements SpanProcessor {
+  private final OnEnd onEnd;
+
+  private OnEndSpanProcessor(OnEnd onEnd) {
+    this.onEnd = onEnd;
+  }
+
+  static SpanProcessor create(OnEnd onEnd) {
+    return new OnEndSpanProcessor(onEnd);
+  }
+
+  @Override
+  public void onEnd(ReadableSpan span) {
+    onEnd.apply(span);
+  }
+
+  @Override
+  public boolean isEndRequired() {
+    return true;
+  }
+
+  @Override
+  public void onStart(Context parentContext, ReadWriteSpan span) {
+    // nop
+  }
+
+  @Override
+  public boolean isStartRequired() {
+    return false;
+  }
+
+  @FunctionalInterface
+  public interface OnEnd {
+    void apply(ReadableSpan span);
```

Comment:
apply or onEnd?

Reply:
I used `apply` for both OnStart and OnEnd. I read it as a functional interface, so an implementation or lambda instance of OnEnd is the function and the function gets applied. At least that's how my brain thinks of it.

I think the name isn't hugely important here, but `OnEnd.onEnd()` seems a little silly. I think one could also make the case that it's a consumer, so it should be called `accept()`, but then `OnStart` has two args.

I guess I think apply is the most straightforward.

Reply:
is this too silly?

```
@FunctionalInterface
public interface OnEndSpanProcessor extends SpanProcessor {

  // this is the kind of silly part...
  static SpanProcessor create(OnEndSpanProcessor onEnd) {
    return onEnd;
  }

  @Override
  default boolean isEndRequired() {
    return true;
  }

  @Override
  default void onStart(Context parentContext, ReadWriteSpan span) {
    // nop
  }

  @Override
  default boolean isStartRequired() {
    return false;
  }
}
```

Reply:
I think that's what I suggested in this [comment](https://github.com/open-telemetry/opentelemetry-java/pull/5923#issuecomment-1771221175) @trask.

Reply:
thx for the link 👍

Reply:
> is this too silly?

It's not, but it does allow people to still do silly things with it. :)